mesonbuild/interpreter/compiler.py

```diff
@@ -320,14 +320,20 @@ def symbols_have_underscore_prefix_method(self, args: T.List['TYPE_var'], kwargs
     @typed_kwargs('compiler.has_member', *_COMMON_KWS)
     def has_member_method(self, args: T.Tuple[str, str], kwargs: 'CommonKW') -> bool:
         typename, membername = args
+        disabled, required, feature = extract_required_kwarg(kwargs, self.subproject, default=False)
```

Comment:
The `required` kwarg must be added into `type_kwargs`, in the same way it is added into `_HEADER_KWS`.

Reply:
help please : /

change this?

```py
# mesonbuild/interpreter/compiler.py
_HEADER_KWS: T.List[KwargInfo] = [REQUIRED_KW.evolve(since='0.50.0', default=False), *_COMMON_KWS]
```

i see `required` only in

```py
# mesonbuild/interpreter/compiler.py
    @disablerIfNotFound
    @typed_pos_args('compiler.find_library', str)
    @typed_kwargs(
        'compiler.find_library',
        KwargInfo('required', (bool, coredata.UserFeatureOption), default=True),
        KwargInfo('has_headers', ContainerTypeInfo(list, str), listify=True, default=[], since='0.50.0'),
        KwargInfo('static', (bool, NoneType), since='0.51.0'),
        KwargInfo('disabler', bool, default=False, since='0.49.0'),
        KwargInfo('dirs', ContainerTypeInfo(list, str), listify=True, default=[]),
        *(k.evolve(name=f'header_{k.name}') for k in _HEADER_KWS)
    )
    def find_library_method(self, args: T.Tuple[str], kwargs: 'FindLibraryKW') -> 'dependencies.ExternalLibrary':
```

Reply:
> _HEADER_KWS: T.List[KwargInfo] = [REQUIRED_KW.evolve(since='0.50.0', default=False), *_COMMON_KWS]

Add a new one:
```python
_HAS_KWS: T.List[KwargInfo] = [REQUIRED_KW.evolve(since='1.1.0', default=False), *_COMMON_KWS]
```

And change `_COMMON_KWS` to `_HAS_KWS` in every `has_*_method`.